doc/architecture/squid.md

```diff
@@ -4,662 +4,709 @@ The Squid API is a Level-2 API built on top of core Ocean components. It's a fac
-metadata (as a service), and other services available with this asset (e.g. Access or Compute service agreements)
+`str local path on the file system where the asset data files are downloaded` 
 
+Example
 ```js
-asset_did = ocean.registerAsset(metadata, publisherAddress, serviceDescriptors)
+const downloadsPath = ocean.assets.consume(agreementId, did, serviceDefinitionId, consumerAccount, mode)
 ```
 
-### signServiceAgreement
+---
+
+## ocean.accounts
 
-ASYNC. Signs the ServiceAgreement from the consumer side
-```
-signature,serviceAgreementId = ocean.signServiceAgreement(did, serviceDefinitionId, consumer)
-```
+| Method              | Return Value                        | Python | JS   | Java |
+| :------------------ | :-----------------------------------| :----- | :--- |:---- |
+| list                | Account list                        |        |      |      |
+| balance             | tuple(ether_balance, ocean_balance) |        |      |      |
+| request_tokens      |                                     |        |      |      |
 
-### executeServiceAgreement
 
-ASYNC. Execute the ServiceAgreement from the publisher side
-```
-ServiceAgreement = ocean.executeServiceAgreement(did, serviceDefinitionId, serviceAgreementId, serviceAgreementSignature, consumer, publisher)
-```
+#### list
 
-### resolveDID
+Returns all available accounts loaded via a wallet, or by Web3.
 
-ASYNC. Given a DID, return the associated DID Document (DDO). The DDO is resolved by directly 
-interacting with the keeper node. The DDO itself is stored off-chain in ocean-db (mongodb, bigchain-db, or whatever)
+Parameters
 
-```js
-DDO = ocean.resolveDID(did)
-```
+Returns
 
-### getOrder
+`list of Account instances`
 
-ASYNC. Get an order from the given orderId.
 
+Example
 ```js
-Order = ocean.getOrder(orderId)
+const accounts = ocean.accounts.list()
 ```
 
-### getAsset
+---
 
-ASYNC. Get an asset based on its identifier, in the form of a DID. 
+#### balance
+Return tuple of both the etherBalance and oceanBalance
 
-```js
-asset = ocean.getAsset(assetDID)
+Parameters
+```
+account: Account instance
 ```
 
-### getServiceAgreement
+Returns
 
-SYNC. Returns as serviceAgreement object based from the service agreement id (Publisher). _Nice to Have_.
+`(etherBalance: float, oceanBalance: int)`
 
+Example
 ```js
-serviceAgreement = ocean.getServiceAgreement(serviceAgreementId)
+const { etherBalance, oceanBalance } = ocean.accounts.balance(accounts[0])
 ```
 
-## Account
+---
+
+#### requestTokens
+This is an alternative to `ocean.tokens.request()`
 
-Ocean Account object.
+Parameters
+```
+account: Account instance
+ amount: int number of tokens to be requested
+```
 
-### getId
+Returns
 
-ASYNC. Return the Id used by this account.
+`boolean`
 
+Example
 ```js
-id= account.getId()
+const amount = 30
+const account = ocean.accounts.list()[0]
+const success = ocean.accounts.requestTokens(account, amount)
 ```
 
-### getOceanBalance
+---
+
+## ocean.secret_store
```

Comment:
Are not private functions?
If are private I'm okay specifying here but detailing are private. 
I would add references to the existing Secret Store documentation in dev-ocea